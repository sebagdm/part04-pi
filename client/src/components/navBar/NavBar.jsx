import React, { useState } from "react";
import s from "./styles/NavBar.module.css";
import logo from "../../img/logo.png";
import { Link } from "react-router-dom";
import { useSelector, useDispatch } from "react-redux";
import {
  findPokemonByName,
  setEn,
  setEs,
  filterBy,
} from "../../redux/actions/actions";
import axios from "axios";
import { useHistory } from "react-router-dom";

const NavBar = () => {
  const dispatch = useDispatch();
  const history = useHistory();
  const color = useSelector((store) => store.navbarBg);
  const [searching, setSearching] = useState(false);
  const [showingResults, setShowingResults] = useState(false);
  const [searchPokemon, setsearchPokemon] = useState("");
  const lang = useSelector((store) => store.lang);
  const findPokemon = () => {
    if (searchPokemon !== "") {
      history.push("/home");
      setSearching(true);
      axios
        .get("/pokemons?name=" + searchPokemon.toLowerCase())
        .then((response) => {
          dispatch(findPokemonByName(response.data));
          setSearching(false);
          setShowingResults(true);
        })
        .catch((e) => {
          if (lang === "en") alert("We couldnt capture this pokemon!");
          if (lang === "es") alert("¡No pudimos capturar ese pokemón!");
          setsearchPokemon("");
          setSearching(false);
        });
    }
  };
  const resetSearch = () => {
    setsearchPokemon("");
    setSearching(false);
    setShowingResults(false);
    dispatch(
      filterBy({
        type: "no_filter",
        showing: "all",
      })
    );
  };

  return (
    <div className={s.nav_shadow} style={{ backgroundColor: `${color}` }}>
      <nav className={s.nav}>
        <Link draggable="false" to="/home">
          <div className={s.logoContainer}>
            <img draggable="false" src={logo} alt="pokemon" />
          </div>
        </Link>
        <div className={s.options}>
          <div className={s.create}>
            <Link to="/home/create">
              <button className={`${s.link}`}>
                <span>
                  {lang === "en"
                    ? "Create your own Pokémons!"
                    : "¡Crea tus Pokémones!"}
                </span>
              </button>
            </Link>
          </div>
          <div className={s.search}>
            <form
              onSubmit={(e) => {
                e.preventDefault();
                findPokemon();
              }}
            >
              <label htmlFor="" className={s.inputContainer}>
                <label
                  htmlFor="search_pokemon"
                  className={searching ? `${s.loadingSearch} ` : s.label}
                >
                  {""}
                </label>
                <input
                  autoComplete="off"
                  type="text"
                  name="search_pokemon"
                  id="search_pokemon"
                  style={{ paddingRight: showingResults ? "25px" : "0px" }}
                  className={s.inputSearch}
                  onChange={(e) => {
                    setsearchPokemon(e.target.value);
                  }}
                  value={searchPokemon}
                  placeholder={
                    lang === "en" ? "Search a pokemon" : "Busca un pokémon"
                  }
                />
                {showingResults ? (
                  <button
                    className={s.closeSearch}
                    onClick={() => {
                      resetSearch();
                    }}
                  >
                    .
                  </button>
                ) : (
                  ""
                )}
              </label>
              <button type="submit" className={`${s.button} ${s.searchBtn}`}>
                <span>{lang === "en" ? "Search" : "Buscar"}</span>
              </button>
            </form>
          </div>
          <div className={s.lang}>
            <button
              className={`${s.button} ${lang === "en" ? s.button_active : ""}`}
              onClick={() => dispatch(setEn())}
            >
              <span>EN</span>
            </button>
            <button
              className={`${s.button} ${lang === "es" ? s.button_active : ""}`}
              onClick={() => dispatch(setEs())}
            >
              <span>ES</span>
            </button>
          </div>
        </div>
      </nav>
    </div>
  );
};

export default NavBar;
