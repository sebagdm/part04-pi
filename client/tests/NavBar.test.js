import "@testing-library/jest-dom";
import Enzyme, { mount } from "enzyme";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import NavBar from "../components/navBar/NavBar";
import { Link } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { BrowserRouter } from "react-router-dom";

Enzyme.configure({ adapter: new Adapter() });

describe("<NavBar />", () => {
  let store;
  let wrapper;
  const mockStore = configureMockStore([thunk]);

  store = mockStore({});
  beforeEach(() => {
    wrapper = mount(
      <BrowserRouter>
        <Provider store={store}>
          <NavBar />
        </Provider>
      </BrowserRouter>
    );
  });

  it("debería mostrar <NavBar />correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Debería haber al menos un <span />", () => {
    expect(wrapper.find(<span />)).toHaveLength(0);
  });

  describe("Verificamos etiquetas de <Link /> en <NavBAr />  ", () => {
    it("Debería tener dos etiquetas <Link />", () => {
      expect(wrapper.find(Link)).toHaveLength(2);
    });
  });
  xdescribe("Probamos las rutas de  <NavBAr /> ", () => {
    it("El primer enlace debería llevar a /home", () => {
      expect(wrapper.find(Link).at(0).prop("to")).toEqual("/home");
    });
    it("Un segundo enlace debería llevar a /home/create", () => {
      expect(wrapper.find(Link).at(1).prop("to")).toEqual("/home/create");
    });
  });
});
