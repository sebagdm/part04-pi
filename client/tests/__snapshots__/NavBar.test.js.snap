// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> deber√≠a mostrar <NavBar />correctamente  1`] = `ReactWrapper {}`;
